### PR TITLE
Update Oct 2023 PSU versions

### PR DESCRIPTION
**Pull request by Luigi96**

### CODE REVIEW

File: .devops/build.yml [hunk 1]
```diff
@@ -39,12 +39,12 @@ parameters:
...
         tag: "image-tag"
-      mariner_cm1_11:
-        distro: mariner-cm1
+      mariner_11:
+        distro: mariner
         version: 11
         package: msopenjdk-11
         image: "image-repository"
-        tag: "1.0"
+        tag: "image-tag"
```
Comment (joe-braley) [inline .devops/build.yml:47]: We can probably have `tag: "2.0"` instead of the placeholder.

Comment (Luigi96) [reply .devops/build.yml]: I updated it for all mariner images

File: .github/workflows/check-versions.yml [hunk 2]
```diff
@@ -28,9 +28,9 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        distros: [ "mariner", "distroless", "mariner-cm1", "ubuntu" ]
```
Comment (jmjaffe37) [inline .github/workflows/check-versions.yml:31]: We don't want to do checks for mariner anymore?

**Comment by jmjaffe37** [comment]
LGTM

**Review by jmjaffe37 — APPROVED** [review]

**Review by d3r3kk — APPROVED** [review]

**Pull request merged**